src/qibolab/platforms/abstract.py

```diff
@@ -325,95 +351,28 @@ def sweep(self, sequence, *sweepers, nshots=1024, average=True, relaxation_time=
                 f"Calibration for CZ gate between qubits {qubits[0]} and {qubits[1]} not found.",
             )
-
-        # If settings contains only one pulse dictionary, convert it into a list that can be iterated below
-        if isinstance(pulse_sequence_settings, dict):
-            pulse_sequence_settings = [pulse_sequence_settings]
-
-        from qibolab.pulses import FluxPulse, PulseSequence
-
-        sequence = PulseSequence()
-        virtual_z_phases = defaultdict(int)
-
-        for pulse_settings in pulse_sequence_settings:
-            if pulse_settings["type"] == "qf":
-                qf_duration = pulse_settings["duration"]
-                qf_amplitude = pulse_settings["amplitude"]
-                qf_shape = pulse_settings["shape"]
-                qubit = pulse_settings["qubit"]
-                if self.qubits[qubit].flux:
-                    qf_channel = self.qubits[qubit].flux.name
-                else:
-                    qf_channel = self.settings["qubit_channel_map"][qubit][2]
-                sequence.add(
-                    FluxPulse(
-                        start + pulse_settings["relative_start"], qf_duration, qf_amplitude, qf_shape, qf_channel, qubit
-                    )
-                )
-            elif pulse_settings["type"] == "virtual_z":
-                virtual_z_phases[pulse_settings["qubit"]] += pulse_settings["phase"]
-            else:
-                raise NotImplementedError(
-                    "Implementation of CZ gates using pulses of types other than `qf` or `virtual_z` is not supported yet."
-                )
-
-        return sequence, virtual_z_phases
+        return self.two_qubit_natives[frozenset(qubits)]["CZ"].sequence(start)
 
     def create_MZ_pulse(self, qubit, start):
-        pulse_kwargs = self.native_single_qubit_gates[qubit]["MZ"]
-        ro_duration = pulse_kwargs["duration"]
-        ro_frequency = pulse_kwargs["frequency"]
-        ro_amplitude = pulse_kwargs["amplitude"]
-        ro_shape = pulse_kwargs["shape"]
-        if self.qubits[qubit].readout:
-            ro_channel = self.qubits[qubit].readout.name
-        else:
-            ro_channel = self.settings["qubit_channel_map"][qubit][0]
-        return ReadoutPulse(start, ro_duration, ro_amplitude, ro_frequency, 0, ro_shape, ro_channel, qubit=qubit)
+        return self.qubits[qubit].native_gates.MZ.pulse(start)
 
     def create_qubit_drive_pulse(self, qubit, start, duration, relative_phase=0):
-        pulse_kwargs = self.native_single_qubit_gates[qubit]["RX"]
-        qd_frequency = pulse_kwargs["frequency"]
-        qd_amplitude = pulse_kwargs["amplitude"]
-        qd_shape = pulse_kwargs["shape"]
-        qd_channel = self.get_qd_channel(qubit)
-        return Pulse(start, duration, qd_amplitude, qd_frequency, relative_phase, qd_shape, qd_channel, qubit=qubit)
+        pulse = self.qubits[qubit].native_gates.RX.pulse(start, relative_phase)
+        pulse.duration = duration
+        return pulse
```

Comment:
/This and following also have a structure similar to the above one, and indeed we could consider something like:
```py
def create_pulse(self, gate: Gate, qubit, attributes: dict, **kwargs):
    natives = self.qubits[qubit].native_gates
    pulse = getattr(natives, gate.name).pulse(**kwargs)
    for name, value in attributes.items():
        setattr(pulse, name, value)
    return pulse
```

But I'd rather avoid doing it, and rather do:
```suggestion
        pulse = self.create_pulse(RX, qubit, start=start, relative_phase=relative_phase)
        pulse.duration = duration
        return pulse
```

Moreover, in principle you should not need to pass `duration` at all to this function, indeed at the moment the caller should be doing something like:
```py
pulse = platform.create_qubit_drive_pulse(qubit, start, duration)
```
but it could do:
```py
pulse = platform.create_qubit_drive_pulse(qubit, start)
pulse.duration = duration
```
The two are not completely equivalent, since the first (and current) one is enforcing the *mandatoriness* of `duration`. However, this is more of a UI detail, but I would consider simplifying the code at the price of making the user more responsible (and in general, functions with fewer arguments are simpler and more composable).

Reply:
> Regarding this and the comment above, the issue is that `platform.create_*_pulse` is part of the qibolab interface and are extensively used in all the routines in qibocal and a few other places. Removing or renaming them will require a major change across two repositories and maybe also break other user scripts.

Ok:
- I see your point, and thus I would also be in favor to decouple from this PR
- we're far from stability, and we have not that many users, so in general is the right time to take this kind of changes into account